src/Servers/Connections.Abstractions/src/ConnectionBuilderExtensions.cs

```diff
@@ -45,6 +45,27 @@ public static IConnectionBuilder Use(this IConnectionBuilder connectionBuilder,
         });
     }
 
+    /// <summary>
+    /// Add the given <paramref name="middleware"/> to the connection.
+    /// If you aren't calling the next function, use <see cref="Run(IConnectionBuilder, Func{ConnectionContext, Task})"/> instead.
+    /// <para>
+    /// Prefer using <see cref="Use(IConnectionBuilder, Func{ConnectionContext, ConnectionDelegate, Task})"/> for better performance as shown below:
```

Comment:
Would it be more helpful to put this suggestion on the non-preferred overload?  I think that's what #31784 did.